index.bs

```diff
@@ -436,22 +439,39 @@ ISSUE: The content of table above needs to be discussed once the profiles and le
 Common Encryption {#CommonEncryption}
 =========================
 
-[=CMAF AV1 Tracks=] and non-segmented AV1 files MAY be encrypted. If encrypted, they SHALL conform to [[!CENC]]. In particular, both <code>[=cenc=]</code> and <code>[=cbcs=]</code> scheme types are permitted.
+[=CMAF AV1 Tracks=] and non-segmented AV1 files MAY be protected. If protected, they SHALL conform to [[!CENC]].
 
-Sample Encryption {#sample-encryption}
---------------------------------------
+When the protected scheme <code>[=cenc=]</code> or <code>[=cbc1=]</code> is used, samples SHALL be protected using subsample encryption and SHALL NOT use pattern encryption.
 
-When encrypting [=OBUs=], all [=OBU Headers=] and associated [=obu_size=] fields SHALL be unencrypted. Additionally, [=Temporal Delimiter OBUs=], [=Sequence Header OBUs=], [=Metadata OBUs=] (except for those requiring protection), [=Frame Header OBUs=] SHALL be unencrypted.
+When the protected scheme <code>[=cens=]</code> or <code>[=cbcs=]</code> is used, samples SHALL be protected using subsample encryption and SHALL use pattern encryption.
 
-Within encrypted samples, all [=OBU Headers=] and associated [=obu_size=] fields SHALL be unencrypted. Additionally, [=Temporal Delimiter OBUs=], [=Sequence Header OBUs=], [=Metadata OBUs=] (except for those requiring protection), [=Frame Header OBUs=] (including with a [=Frame OBU=]) SHALL be unencrypted. [=Tile Group OBUs=], [=Frame OBUs=] (except the [=Frame Header OBU=] part) and [=Tile List OBUs=] MAY be encrypted. This is illustrated in Figure #1 and Figure #2, where for simplicity the [=obu_size=] field is assumed to be part of the [=OBU Header=].
+General Subsample Encryption constraints {#subsample-encryption}
+--------------------------------------
 
-As permitted by [[!CENC]], multiple unencrypted OBUs MAY be described by a single subsample. Also, multiple subsamples MAY be used to describe the data of a single OBU.
+Within protected samples, the following constraints apply:
+- Protected samples SHALL be exactly spanned by one or more contiguous subsamples.
+- all [=OBU Headers=] and associated [=obu_size=] fields SHALL be unprotected.
+- [=Temporal Delimiter OBUs=], [=Sequence Header OBUs=], and [=Frame Header OBUs=] SHALL be unprotected.
+- [=Metadata OBUs=] MAY be protected.
+- [=Tile Group OBUs=], [=Frame OBUs=], and [=Tile List OBUs=] SHALL be protected.
+- Within [=Tile Group OBUs=] or [=Frame OBUs=], only the bytes after <code>byte_alignment</code> in the <code>tile_group_obu</code> structure SHALL be protected.
+- Within [=Tile List OBUs=], only the <code>coded_tile_data</code> part SHALL be protected.
+- [=BytesOfProtectedData=] size SHALL be a multiple of 16 bytes to avoid partial cipher blocks in subsamples.
```

Comment:
CENC 9.5.1 says:
>For ‘cenc’ and ‘cens’ protection schemes, BytesOfProtectedData SHOULD be adjusted to a
multiple of 16 bytes to avoid partial Blocks at the end of Subsamples.

CENC 10.1 'cenc' says:
> The BytesOfProtectedData size SHOULD be a multiple of 16 bytes to avoid partial cipher blocks
in Subsamples.

CENC 10.3 'cens' says:
> The BytesOfProtectedData size SHALL be a multiple of 16 bytes to avoid partial cipher blocks
in Subsamples.

There is some inconsistency in CENC here but I guess the stricter rule applies. I'll forward that to MPEG (ping @dwsinger).

CMAF says:
>- Video VCL data shall be protected by subsample encryption, with
bytesOfProtectedData spanning all complete 16-byte blocks in the VCL slice data.
>- 'cenc' scheme bytesOfProtectedData shall be a multiple of 16 bytes